### DISCUSSION TITLE
How to encrypt assets with nuxt3?

### DISCUSSION THREAD

**Discussion by Oskilla69** [nuxt/nuxt#22493] [General]
Hi,
I'm currently working on a project where my assets need to be in the public directory so that they are dynamically accessible by my code. However, these assets are sensitive and SSR isn't an option in my case, so I want to encrypt them. How would I go about doing that?

I'm new to encryption, so if this is not the right place to ask such a question, I'd greatly appreciate it if you could point me to the right place. Thank you so much.

**Comment by TheAlexLichter** [comment]
As soon as your assets are available the frontend, encryption doesn't matter much as all the source code is publicly readable, as well as a possible encryption key. 

Thus, frontend-only encryption is not a secure option.

This is also framework-agnostic and not nuxt-related

**Reply by Oskilla69** [reply]
Hey Alex, thank you so much for answering my question. If that is the case, is there anything I can do?

**Reply by TheAlexLichter** [reply]
I think not much except getting your data via SSR or from an API.

**Reply by Oskilla69** [reply]
How would getting data from an API be any different? Couldn't they just read the assets from the network requests?

**Reply by TheAlexLichter** [reply]
Yes, but if they are user-bound, you can only emit what is necessary to each user ☺️

**Reply by Oskilla69** [reply]
Oh yep makes sense. Thank you so much for answering my questions.
